src/signature.c

```diff
@@ -20,7 +21,84 @@ void signature_init(void)
 	ERR_load_crypto_strings();
 }
 
-static EVP_PKEY *load_key(const gchar *keyfile, GError **error)
+static ENGINE *get_pkcs11_engine(GError **error)
```

Comment:
Please add a `g_return_val_if_fail(..)` for `error`

Reply:
Done.